src/index.js

```diff
@@ -44,10 +44,10 @@ export default function transformCssModules({ types: t }) {
     // is css modules require hook initialized?
     let initialized = false;
 
-    let matchExtensions = /\.css/i;
+    let matchExtensions = /\.css$/i;
     function matcher(extensions = ['.css']) {
-        const extensionsPatern = extensions.join('|').replace('.', '\.');
-        return new RegExp(`(${extensionsPatern})`, 'i');
+        const extensionsPatern = extensions.join('|').replace(/\./g, '\\\.');
+        return new RegExp(`(${extensionsPatern})$`, 'i');
```

Comment:
If we're anchoring the match at the end of a stylesheet path (which I think we should), we might want to take into account possible url parameters too... thoughts?

We could use something like this instead:

```javascript
return new RegExp(`(${extensionsPatern})(\?.*)?$`, 'i');
```